docs/reference/networking.md

```diff
@@ -1,13 +1,89 @@
 # Virtlet networking principles
 
+The only supported configuration is the use of
+[CNI plugins](https://github.com/containernetworking/cni). Virtlet should work
+with any CNI plugin which is returning correct
+[Result](https://github.com/containernetworking/cni/blob/spec-v0.3.1/SPEC.md#result)
+in at least 0.3.0 format, according to CNI protocol specification.
+
+Virtlet assumes that selected CNI plugins will plug prepared for VM network
+namespace into own network using [veth](http://man7.org/linux/man-pages/man4/veth.4.html)
+or [SR-IOV](https://en.wikipedia.org/wiki/Single-root_input/output_virtualization)
+interfaces.
+
+For each configured veth interface Virtlet will prepare additional
+bridge (used later by internal DHCP server) used to connect this veth with
+[TAP](https://en.wikipedia.org/wiki/TUN/TAP) which will be then passed to
+hypervisor ([QEMU](https://www.qemu.org)) process.
+
+Each configured SR-IOV device will be removed from host kernel configuration
+and passed to hypervisor as host-passthrough PCI device preserving their
+hardware MAC addresses and optionanlly configured by plugin [VLAN](https://en.wikipedia.org/wiki/Virtual_LAN).
+
 # Supported CNI implementations
 
+Virtlet was tested and is working correctly with:
+
+* [Flannel](https://github.com/coreos/flannel)
+* [Calico](https://github.com/projectcalico/cni-plugin)
+* [Weave](https://github.com/weaveworks/weave)
+* [original](https://github.com/hustcat/sriov-cni) and [Intel fork](https://github.com/intel/sriov-cni) of SR-IOV
+* [combination](#multi-cni) of them behind [CNI-Genie](https://github.com/Huawei-PaaS/CNI-Genie)
+
 # Configuring using DHCP
 
+Information collected from CNI plugin is used by DHCP server accessible on each
+of configured for VM interfaces. That data is visible only for particular VM
+and is limited to serve configuration for particular interface, per interface.
+
+DHCP configuration is unavailable for [SR-IOV](#sriov) devices as they are
+passed directly to VM while fully excluded from host kernel configuration.
+
 # Configuring using Cloud-Init
 
-Not supported for persistent rootfs for now
+In parallel to passing network configuration through DHCP it is also passed to
```

Comment:
replace `In parallel to passing network configuration through` with 'In addition to passing network configuration through`